File: nodes/model_selection/train_test_split/train_test_split.py
import pandas as pd
import sys
import json
import numpy as np

from utils.utils import myprint as print
from utils.node import Node

from sklearn.model_selection import train_test_split as ttsplit


# ? Random state de input olarak verilebilir mi?
class Split(Node):
    def __init__(self, *args):
        super().__init__(*args)
        self.inputs = [Data]
        self.type = Node.Type.DATA
    
    def function(self, X, y, onlyTest, testPercentage):
        X_train, X_test, y_train, y_test = ttsplit(X, y, test_size=testPercentage / 100)
        return [(X_train, y_train, onlyTest), (X_test, y_test, True)]
        #      * Use X_train for train depending on the configuraiton
        #      * Use X_test only for testing
    

File: nodes/model_selection/train_test_split/train_test_split.js
module.exports = function(RED) {
    function TrainTestSplitNode(config) {

        const utils = require('../../../utils/utils');

        var node = this;

        node.topic = 'data';
        // Corresponding python class
        node.pynode = 'Split';

        node.config = {
            testPercentage: parseInt(config.testPercentage)
        };

        utils.run(RED, node, config);
    }
    RED.nodes.registerType('train test split', TrainTestSplitNode);
}